docs/bbe-guidelines/bbe-guidelines.md

```diff
@@ -0,0 +1,122 @@
+
+    For example, it will not be readable if comments are significantly longer than the code line as shown below. 
+
+    ![long comments](images/long-comments.png "long-comments")
+
+## Practices related to the folder structure
+
+1. Each BBE should be in a separate directory in the [ballerina-lang Git Hub repository](https://github.com/ballerina-platform/ballerina-lang/tree/master/examples) named by the example (e.g., https://github.com/ballerina-platform/ballerina-lang/tree/master/examples/artemis-transaction-producer).
+
+2. Directory name should be in all lowercase letters with words separated by “-” (e.g., abstract-objects).
+
+3. Each example should contain at least the following files. 
+
+    > **Tip:** File names should be the same as folder name, with the “-” being replaced with the underscore (e.g., abstract_objects).
+
+    `.bal` - The code of the example to be displayed in the [Ballerina By Example](https://ballerina.io/learn/by-example/).
+
+    `.description` - The description of the example displayed at the top of each example in the [Ballerina By Example](https://ballerina.io/learn/by-example/).
+
+    `.out` - Output of the example displayed at the bottom black colour box in the [Ballerina By Example](https://ballerina.io/learn/by-example/).
+
+    `<example_name>_test.bal` - Contains the test to validate the output of the BBE during the build time. 
+
+    <img src="images/bbe-folder-structure.png" width="300" height="150" alt="BBE folder structure" />
+
+4. As a common pattern, use the following format for `.out` files and customize it only when required (e.g., to add more command line args etc.).
+
+- For an example with main;
+
+    ```
+    # To run this sample, navigate to the directory that contains the
+    # `.bal` file and execute the `ballerina run` command.
+    $ ballerina run <sample_file_name>.bal
+    ```
+- For an example with a service;
+
+    ```
+    # To start the service, navigate to the directory that contains the
+    # `.bal` file and execute the `ballerina run` command.
+    $ ballerina run hello_world_service.bal
+    ```
+
+
+5. Service examples need two `.out` files. One (i.e., the `server.out` file) to display the server output and the other (i.e., the `client.out` file) to display the cURL command and the output.
+
+    <img src="images/service-example-structure.png" width="330" height="180" alt="service example structure" />
+
+
+6. Unless it is really required, it is not encouraged to have multiple `.bal` files in the same example. In case if multiple are needed, each BAL file can have its own name and the respective `.out` files should match the name of the respective `.bal` file. 
```

Comment:
```suggestion
6. Unless it is really required, it is not encouraged to have multiple `.bal` files in the same example. In case if multiple are needed, each `.bal` file can have its own name and the respective `.out` files should match the name of the respective `.bal` file. 
```

Reply:
Fixed.